### PR TITLE
[Schema] Add password masking to the 1.5 schema

### LINKED ISSUE #6339: Input.Text masking/password Schema support

### LINKED ISSUE #6343: Update schema for official 1.5 version
Update the schema version for 1.5 and verify new features are v1.5 and not vNext

### PR DESCRIPTION
**Pull request by paulcam206**
# Related Issue

Fixes #6343

# Description

Adds schema enum value versioning support (already supported by our markdown generator) and introduces the `password` style for `Input.Text`.

# How Verified

* local build of website

image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6376)

### CODE REVIEW

**Comment by RebeccaAnne** [comment]
I think usually we copy the generated schema over into AdaptiveCards\schemas\1.5.0\adaptive-card.json as part of this type of change. Also, are the specs in AdaptiveCards\specs\elements updated and checked in automatically?

**Comment by paulcam206** [comment]
> I think usually we copy the generated schema over into AdaptiveCards\schemas\1.5.0\adaptive-card.json as part of this type of change.

Ooh! Good to know. I'll get that updated. It occurs to me that I've not made a schema change in quite some time...

> Also, are the specs in AdaptiveCards\specs\elements updated and checked in automatically?

I'm not sure -- I'll check.

File: source/nodejs/ac-typed-schema/src/SchemaEnumValue.ts [hunk 1]
```diff
@@ -27,7 +29,11 @@ export class SchemaEnumValue {
 		return this._description;
 	}
 
+    get version() {
+        return this._version;
+    }
+
 	get original() {
 		return this._original;
```
Comment (paulcam206) [inline source/nodejs/ac-typed-schema/src/SchemaEnumValue.ts:37]: some whitespace issues here

**Comment by paulcam206** [comment]
> > Also, are the specs in AdaptiveCards\specs\elements updated and checked in automatically?
> 
> I'm not sure -- I'll check.

Closing the loop here, it looks like PR builds will update and check in automatically, except spec generation was broken (due to a missing `}` in [`host-config.json`](https://github.com/microsoft/AdaptiveCards/blob/main/schemas/host-config.json) and some other issues I'm looking at).

File: schemas/host-config.json [hunk 2]
```diff
@@ -685,64 +674,73 @@
...
+				"allowCustomStyle": {
+					"type": "boolean",
+					"description": "Controls whether custom styling is allowed",
+					"default": true
+				},
+				"actions": {
+					"$ref": "#/definitions/ActionsConfig"
+				},
+				"adaptiveCard": {
+					"$ref": "#/definitions/AdaptiveCardConfig"
```
Comment (RebeccaAnne) [inline schemas/host-config.json:705]: This is referring back to it's own parent object, which doesn't seem right. In our samples there is an AdaptiveCard property under the top level host config that only has allowCustomStyle in it:

{
...
    "adaptiveCard": {
        "allowCustomStyle": false
     },
...
}

File: source/nodejs/ac-typed-schema/src/markdown/languages/de.json [hunk 3]
```diff
@@ -219,5 +219,7 @@
...
-	"Defines how the content should be aligned vertically within the column. When not specified, the value of verticalContentAlignment is inherited from the parent container. If no parent container has verticalContentAlignment set, it defaults to Top.": "Defines how the content should be aligned vertically within the column. When not specified, the value of verticalContentAlignment is inherited from the parent container. If no parent container has verticalContentAlignment set, it defaults to Top."
+	"Defines how the content should be aligned vertically within the column. When not specified, the value of verticalContentAlignment is inherited from the parent container. If no parent container has verticalContentAlignment set, it defaults to Top.": "Defines how the content should be aligned vertically within the column. When not specified, the value of verticalContentAlignment is inherited from the parent container. If no parent container has verticalContentAlignment set, it defaults to Top.",
+	"Display input text with password masking": "Display input text with password masking",
+	"Display input text with password masking.": "Display input text with password masking."
```
Comment (RebeccaAnne) [inline source/nodejs/ac-typed-schema/src/markdown/languages/de.json:224]: This is in here twice now.

File: schemas/host-config.json [hunk 4]
```diff
@@ -613,7 +602,7 @@
 				}
 			}
 		},
-		"TextStyleConfig": {
+		"TextStylesConfig": {
```
Comment (RebeccaAnne) [inline schemas/host-config.json:605]: >TextStylesConfig [](http://example.com/codeflow?start=3&length=16)

I think this is actually TextStyleConfig

File: schemas/host-config.json [hunk 5]
```diff
@@ -669,12 +658,12 @@
 				}
 			}
 		},
-		"TextStylesConfig": {
+		"TextBlockConfig": {
```
Comment (RebeccaAnne) [inline schemas/host-config.json:661]: >TextBlockConfig [](http://example.com/codeflow?start=3&length=15)

This should be TextStylesConfig, and should include a TextStyleConfig for each of "heading" and "columnHeader"

TextBlock config should also exist, it contains only "headingLevel"

(per shared model code)

File: specs/elements/ActionSet.md [hunk 6]
```diff
@@ -45,7 +45,7 @@ Describes what to do when an unknown element is encountered or the requires of t
 * **Allowed values**:
   * `ActionSet`
   * `ColumnSet`
-  * `Container`
```
Comment (RebeccaAnne) [inline specs/elements/ActionSet.md:48]: >Container [](http://example.com/codeflow?start=5&length=9)

This is almost certainly not caused by your change, but I'm curious what happened that made Container fall off this list. Also, TableCell and TableRow aren't really fallback options for an ActionSet.

Comment (paulcam206) [reply specs/elements/ActionSet.md:48]: yeah - I'm in the middle of tracking some of these issues down :)

**Comment by paulcam206** [comment]
@almedina-ms - just a heads up... not sure if you knew about the spec generation pipelines ([here](https://microsoft.visualstudio.com/AdaptiveCards/_build?definitionId=40932&_a=summary) and [here](https://microsoft.visualstudio.com/AdaptiveCards/_build?definitionId=40939&_a=summary)). I certainly did not... at any rate, I got them unblocked and working correctly again (theoretically), but the functionality should probably get baked in to the overall NodeJS build job. I'm not sure if that's an area you're working in, just something to be aware of 👍

**Review by RebeccaAnne — APPROVED** [review]

File: schemas/README.md [hunk 7]
```diff
@@ -15,7 +19,11 @@ The Adaptive Card schema is generated from "typed schema" files located in the `
 
 Our schema explorer on the website is also generated from the typed schema files in the `src` folder.
 
-To generate the schema, see `../source/nodejs/ac-typed-schema`.
+To generate the schema, see [ac-typed-schema](../source/nodejs/ac-typed-schema). Once built (see the nodejs [README](../source/nodejs/README.md)), run `npm run generate-adaptive-schema`. This will output the generated schema as `schemas/adaptive-card.json` (which should not be checked in). From here, the file should be copied into the version-appropriate subfolder (e.g. `cd schemas && cp adaptive-card.json 1.4.0/adaptive-card.json`).
+
+### Generating the schema spec markdown
+
+Once the schema itself is generating correctly, the markdown specs need to be generated as well, at least for testing (the markdown files are automatically generated by a PR builds). The tool you need is [spec-generator](../source/nodejs/spec-generator). Once built, issue the command `npm run run` (yes, really), which will generate the various markdown files you can find in the [specs folder](../specs). More details can be found in the specs README.
```
Comment (almedina-ms) [inline schemas/README.md:26]: >(yes, really) [](http://example.com/codeflow?start=299&length=13)

love the commentary<3 but probably explaining the second run is the name of the task can also be helpful to explain why run run

File: specs/HostConfig.md [hunk 8]
```diff
@@ -18,10 +19,22 @@
...
+|**textBlock**|`object`| No|Configuration settings for TextBlocks|1.0
+|**textStyles**|`object`| No|Sets default properties for text of a given style|1.0
+
+
+
+## AdaptiveCardConfig
+
+|Property|Type|Required|Description|Version|
+|--------|----|--------|-----------|-------|
+|**allowCustomStyle**|`boolean`| No, default: `true`|Controls whether custom styling is allowed|1.0
```
Comment (almedina-ms) [inline specs/HostConfig.md:31]: >1.0 [](http://example.com/codeflow?start=96&length=3)

missing "|" character at the end

**Review by almedina-ms — APPROVED** [review]

File: schemas/README.md (hunk 7 above)
Comment (paulcam206) [reply schemas/README.md:26]: I'll fix it up in the next PR :)

**Pull request merged**